### PR TITLE
HEP for vgpu support [CI SKIP]

### PR DESCRIPTION
**Pull request by ibrokethecloud**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

HEP to represent for vGPU devices support in pcidevices controller

**Solution:**

**Related Issue:**
https://github.com/harvester/harvester/issues/2764
**Test plan:**

### CODE REVIEW

File: enhancements/20231207-vgpu-support.md [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+- Allow users to enable / disable vGPU on NVIDIA GPUs
+
+- Configure vGPU using available profiles based on underlying GPU type
+
+- Allow usage of vGPU in VM workloads
+
+### Non-goals [optional]
+
+Shipping NVIDIA drivers.
```
Comment (bk201) [inline enhancements/20231207-vgpu-support.md]: license management in guests is also a non-goal.

File: enhancements/20231207-vgpu-support.md [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+### API changes
+No API changes will be introduced to core Harvester. The additional CRDs will be managed by the PCI devices controller, as a result the end users will need to enable the PCI devices controller addon to leverage this capability.
+
+In addition we will also be shipping a `nvidia-driver-toolkit` addon. Users need to enable this addon, and provide info about NVIDIA driver location for driver to be installed at runtime.
+
+## Design
+
+The controller introduces two new CRDs:
+- sriovgpudevices.devices.harvesterhci.io
+- vgpdevices.devices.harvesterhci.io
```
Comment (torchiaf) [inline enhancements/20231207-vgpu-support.md]: Is this one `vgpudevices.devices.harvesterhci.io` ?

Comment (ibrokethecloud) [reply enhancements/20231207-vgpu-support.md]: yep.

File: enhancements/20231207-vgpu-support.md [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+A Harvester user wants to leverage vGPU capability on supported NVIDIA GPU devices and share vGPUs with VM workloads.
+
+Currently configuring and passing through vGPU devices in a cluster is not possible as the user can only passthrough the entire GPU to the VM workload. This can result in less efficient resource utilisation of GPU instances.
+
+Once this change is implemented, the user will be able to browse SR-IOV capable NVIDIA GPU devices on the cluster. The user can then enable / disable vGPUs on said GPU. The newly created vGPU devices will appear as a new CRD, with information about all the possible vGPU profiles supported by the vGPU. 
+
+The user can now enable / disable vGPUs by defining the correct profile available from list of possible profiles. Once the vGPU is setup, it can be passed through to VM for use by workloads.
+
+### API changes
+No API changes will be introduced to core Harvester. The additional CRDs will be managed by the PCI devices controller, as a result the end users will need to enable the PCI devices controller addon to leverage this capability.
```
Comment (connorkuehl) [inline enhancements/20231207-vgpu-support.md:45]: Are we able to express dependencies between harvester addons? I.e., automatically enable the PCI devices addon (if it is not already) when the vGPU addon is enabled.

Comment (ibrokethecloud) [reply enhancements/20231207-vgpu-support.md:45]: we could, i wrote an initial PR for this: https://github.com/harvester/harvester/pull/3006

I am not so sure if we need dependencies yet.

**Review by bk201 — APPROVED** [review]

**Review by noahgildersleeve — APPROVED** [review]
LGTM

**Review by connorkuehl — APPROVED** [review]

**Review by torchiaf — APPROVED** [review]
LGTM

**Pull request merged**